### PR TITLE
MON-21239-Complete notifications to centreon-tests--draft

### PR DESCRIPTION
**Pull request by sfarouq-ext**
## Description

Please include a short resume of the changes and what is the purpose of PR. Any relevant information should be added to help:
* **QA Team** (Quality Assurance) with tests.
* **reviewers** to understand what are the stakes of the pull request.

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [x] 23.10.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

## Checklist

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: tests/resources/Engine.py [hunk 1]
```diff
@@ -2206,8 +2293,28 @@ def config_host_command_status(idx: int, cmd_name: str, status: int):
...
 
     with open(filename, "w") as f:
         f.writelines(lines)
+
+def set_service_state(id: int, state: int):
+    # Read the contents of the file and split it into lines
+    with open('/tmp/states', 'a+') as f:
+        lines = f.readlines()
+
+    r = re.compile(rf"^{id}=>[0-9]+")
```
Comment (thearchit3ct) [inline tests/resources/Engine.py:2307]:
```suggestion
    r = re.compile(rf"^{id}=>[0-9]+")
```

File: tests/resources/Engine.py [hunk 2]
```diff
@@ -629,6 +695,9 @@ def build_configs(self, hosts: int, services_by_host: int, debug_level=0):
             f = open(config_dir + "/escalations.cfg", "w")
             f.close()
 
+            f = open(config_dir + "/meta_services.cfg", "w")
```
Comment (thearchit3ct) [inline tests/resources/Engine.py:698]:
```suggestion
            f = open(f"{config_dir}/meta_services.cfg", "w")
```

File: tests/broker-engine/notifications.robot [hunk 3]
```diff
@@ -0,0 +1,1490 @@
...
+    # Let's wait for the external command check start
+    ${content}    Create List    check_for_external_commands()
+    ${result}    Find In Log with Timeout    ${engineLog0}    ${start}    ${content}    60
+    Should Be True    ${result}    check_for_external_commands() should be available.
+
+    ## Time to set the service3 to CRITICAL HARD.
+    FOR   ${i}    IN RANGE    ${4}
+        Process Service Result Hard    host_3    service_3    ${2}    The service_3 is CRITICAL
+        Sleep    1s
+    END
```
Comment (thearchit3ct) [inline tests/broker-engine/notifications.robot:1005]:
```suggestion
        Sleep    1s
```

File: tests/broker-engine/notifications.robot [hunk 4]
```diff
@@ -0,0 +1,1490 @@
...
+
+    ${content}    Create List    SERVICE NOTIFICATION: John_Doe;host_3;service_3;CRITICAL;command_notif;
+    ${result}    Find In Log with Timeout    ${engineLog0}    ${start}    ${content}    60
+    Should Be True    ${result}    The notification is not sent for service3
+
+    ## Time to set the service3 to OK hard
+    FOR   ${i}    IN RANGE    ${3}
+        Process Service Result Hard    host_3    service_3    ${0}    The service_3 is OK
+            Sleep    1s
+    END
```
Comment (thearchit3ct) [inline tests/broker-engine/notifications.robot:1020]:
```suggestion
        Sleep    1s
```

File: tests/broker-engine/notifications.robot [hunk 5]
```diff
@@ -0,0 +1,1490 @@
...
+
+    ${content}    Create List    SERVICE NOTIFICATION: John_Doe;host_3;service_3;RECOVERY (OK);command_notif;
+    ${result}    Find In Log with Timeout    ${engineLog0}    ${start}    ${content}    60
+    Should Be True    ${result}    The notification is not sent for service3
+
+    ## Time to set the service1 to CRITICAL HARD.
+    FOR   ${i}    IN RANGE    ${4}
+        Process Service Result Hard    host_1    service_1    ${2}    The service_1 is CRITICAL
+        Sleep    1s
+    END
```
Comment (thearchit3ct) [inline tests/broker-engine/notifications.robot:1033]:
```suggestion
        Sleep    1s
```

File: tests/broker-engine/notifications.robot [hunk 6]
```diff
@@ -0,0 +1,1490 @@
...
+    Stop Engine
+    Kindly Stop Broker
+
+
+not16
+    [Documentation]    several notification commands for the same user.
+    [Tags]    broker    engine    services    unified_sql
+    Config Engine    ${1}    ${1}    ${1}
+    Config Notifications
+    engine_config_add_command
```
Comment (thearchit3ct) [inline tests/broker-engine/notifications.robot:898]:
```suggestion
    Engine Config Add Command
```

File: tests/broker-engine/notifications.robot [hunk 7]
```diff
@@ -0,0 +1,1490 @@
...
+    Start Broker
+    Start Engine
+
+    # Let's wait for the external command check start
+    ${content}    Create List    check_for_external_commands()
+    ${result}    Find In Log with Timeout    ${engineLog0}    ${start}    ${content}    60
+    Should Be True    ${result}    A message telling check_for_external_commands() should be available.
+
+    # Time to set the service to CRITICAL HARD.
+
```
Comment (thearchit3ct) [inline tests/broker-engine/notifications.robot:923]:
```suggestion
    ${content}    Create List    check_for_external_commands()
```

Comment (thearchit3ct) [inline tests/broker-engine/notifications.robot:923]:
```suggestion
    Should Be True    ${result}    A message telling check_for_external_commands() should be available.
```

File: tests/broker-engine/notifications.robot [hunk 8]
```diff
@@ -0,0 +1,1490 @@
...
+    ${result}    Find In Log With Timeout    ${engineLog0}    ${start}    ${content}    90
+    Should Be True    ${result}    check_for_external_commands() should be available.
+
+    ## Time to set the service to CRITICAL HARD.
+
+    Process Service Result Hard    host_1    service_1    ${2}    The service_1 is CRITICAL
+
+    ## Time to set the service to CRITICAL HARD.
+    Set Service State    ${38}    ${2}
+
```
Comment (thearchit3ct) [inline tests/broker-engine/notifications.robot:94]:
```suggestion
    Set Service State    ${38}    ${2}
```

File: tests/resources/engine-scripts/check.pl [hunk 9]
```diff
@@ -2,42 +2,56 @@
 
 use strict;
 use warnings;
+use Getopt::Long;
```
Comment (jean-christophe81) [inline tests/resources/engine-scripts/check.pl:5]: whitelist.robot is not updated

**Comment by sfarouq-ext** [comment]
the actual PR [Mon 21239 new notification test1](https://github.com/centreon/centreon-collect/pull/1314)

**Pull request closed without merging**